lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,346 @@
+            let _ = std::fs::create_dir_all(parent);
+        }
+
+        let cargo_wapm_stdout = std::process::Command::new("cargo")
+            .arg("wapm")
+            .arg("--version")
+            .output()
+            .map(|s| String::from_utf8_lossy(&s.stdout).to_string())
+            .unwrap_or_default();
+
+        let cargo_wapm_present =
+            cargo_wapm_stdout.lines().count() == 1 && cargo_wapm_stdout.contains("cargo wapm");
+        let should_add_to_cargo_toml = cargo_toml.is_some() && cargo_wapm_present;
+
+        if !cargo_wapm_present && cargo_toml.is_some() {
+            eprintln!(
+                "Note: you seem to have a Cargo.toml file, but you haven't installed `cargo wapm`."
+            );
+            eprintln!("You can build and release Rust projects directly with `cargo wapm publish`: https://crates.io/crates/cargo-wapm");
+            eprintln!("Install it with:");
+            eprintln!("    cargo install cargo-wapm");
+            eprintln!();
+        }
+
+        if should_add_to_cargo_toml {
+            let toml_string = toml::to_string_pretty(&default_manifest)?;
+            let mut value = toml_string.parse::<toml::Value>().unwrap();
+            value.as_table_mut().unwrap().remove("package");
+
+            let mut new_table = toml::value::Table::new();
+            new_table.insert("wapm".to_string(), value);
+
+            let mut table_2 = toml::value::Table::new();
+            table_2.insert("metadata".to_string(), new_table.into());
+
+            let toml_string = toml::to_string_pretty(&table_2)?;
+
+            let cargo_toml_path = target_file.parent().unwrap().join("Cargo.toml");
+
+            let old_cargo = std::fs::read_to_string(&cargo_toml_path).unwrap();
+
+            if old_cargo.contains("metadata.wapm") {
+                return Err(anyhow::anyhow!(
+                    "wapm project already initialized in Cargo.toml file"
+                ));
+            } else {
+                eprintln!("You have cargo-wapm installed, added metadata to Cargo.toml instead of wapm.toml");
+                eprintln!("Build and publish your package with:");
+                eprintln!();
+                eprintln!("    cargo wapm publish");
+                eprintln!();
+                std::fs::write(
+                    &cargo_toml_path,
+                    &format!("{old_cargo}\r\n\r\n# See more keys and definitions at https://docs.wasmer.io/ecosystem/wapm/manifest\r\n\r\n{toml_string}"),
+                )?;
+            }
+        } else {
+            let toml_string = toml::to_string_pretty(&default_manifest)?
+            .replace("[dependencies]", "# See more keys and definitions at https://docs.wasmer.io/ecosystem/wapm/manifest\r\n\r\n[dependencies]")
+            .lines()
+            .collect::<Vec<_>>()
+            .join("\r\n");
+
+            std::fs::write(target_file, &toml_string)?;
+        }
+
+        Ok(())
+    }
+}
```

Comment:
Are you able to refactor `Init::execute()` and introduce some higher-level abstractions?

I'm not trying to be pedantic here, there's lots of stuff going on with lots of implicit assumptions and it's not immediately apparent what `wasmer init` will actually generate.

For example, if people can choose templates then I'd expect there to be separate functions for setting up JavaScript and Python. Or possibly some `Template` struct which is populated with the source for Python/JS-specific files.

Similarly, we spend a lot of lines special-casing `Cargo.toml`, so that should be split out into its own set of functions for retrieving information from `cargo` and constructing the `wapm_toml::Manifest`.